### PR TITLE
Adding Debugging section into Polycube Developers Guide

### PR DESCRIPTION
**Pull request by michelsciortino**
This PR adds a new Debugging section into Polycube Developer Guide:

"Debugging and logging in the data plane" and "Debugging and logging in the control plane" sections have been brought together in a single document; this document explains the use of polycube logging features and shows how to filter the log messages produced by the polycubed-daemon and running services.

### CODE REVIEW

File: Documentation/developers/debugging.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+::
+
+  pcn_log(ctx, LOG_DEBUG, "Receiving packet from port %d", md->in_port);
+
+
+The ``pcn_pkt_log(ctx, level)`` primitive sends a packet to the control plane where it is printed in a tcpdump-like format.
+`ctx` and `level` are the same as in `pcn_log`.
+This feature is only designed for developers, so final version of services should not include this.
+
+`polycube-tools <https://github.com/mauriciovasquezbernal/polycube-tools>`_ must be installed and ``polycube`` cmake must be configured with `cmake -DHAVE_POLYCUBE_TOOLS=ON ..`
```
Comment (frisso) [inline Documentation/developers/debugging.rst]: @mauriciovasquezbernal Is this still valid? As far as I know, @michelsciortino was not able to make it working and he didn't understand exactly what this was about.

Comment (frisso) [reply Documentation/developers/debugging.rst]: Thanks, Mauricio.
@michelsciortino can you please add a new commit in which you:
1) update the documentation, removing all the references to polycube-tools;
2) update the code, as the "HAVE_POLYCUBE_TOOLS" macro is present in many makefiles and it should be removed.

Comment (goldenrye) [reply Documentation/developers/debugging.rst]: @frisso so this means logging with packet dumping in control plane not supported any more or there is alternatives?

Comment (frisso) [reply Documentation/developers/debugging.rst]: You can use the new "span" mode for a port, which enables to receive all packets on a new interface available in a dedicated namespace. There, you can use standard tools such as tcpdump/Wireshark to inspect the traffic.

Comment (frisso) [reply Documentation/developers/debugging.rst]: @michelsciortino You should re-organize the commits (using the git squash function) in order to make it easier to understand. Details by voice.

Comment (goldenrye) [reply Documentation/developers/debugging.rst]: @michelsciortino what's the status of this PR? Is the re-base done?

**Pull request merged**